src/client/containers/LandingPage/LandingPage.container.js

```diff
@@ -1,15 +1,79 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './LandingPage.styles.css';
-import { useProducts } from '../../hooks/useProducts';
-import Products from '../../components/Products/Products.component';
+import { useProducts, PRODUCT_PER_PAGE } from '../../hooks/useProducts';
+import Loader from '../../components/Loader/Loader.component';
+import Sort from '../../components/Sort/Sort.component';
+import Pagination from '../../components/Pagination/Pagination.component';
+import SpecialOfferBanner from '../../components/SpecialOfferBanner/SpecialOfferBanner.component';
+import ProductsView from '../../components/ProductsView/ProductsView.component';
 
 export default function LandingPage() {
-  const { products, isLoading, error } = useProducts();
+  const [sortingPreferences, setSortingPreferences] = useState({
+    sortKey: 'name',
+    sortOrder: 'asc',
+  });
+  const { isLoading, error, products, totalCount, loadMoreProducts } =
+    useProducts(sortingPreferences);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const maxNumberOfPages = Math.ceil(totalCount / PRODUCT_PER_PAGE);
+
+  const pages = [];
+  for (let i = 1; i <= Math.ceil(products.length / PRODUCT_PER_PAGE); i += 1) {
+    pages.push(i);
+  }
+
+  const getCurrentProducts = (page) => {
+    const indexOfLastProduct = page * PRODUCT_PER_PAGE;
+    const indexOfFirstProduct = indexOfLastProduct - PRODUCT_PER_PAGE;
+    const currentProductsOnPage = products.slice(
+      indexOfFirstProduct,
+      indexOfLastProduct,
+    );
+    return currentProductsOnPage;
+  };
+
+  const handlePagination = (pageNumber) => {
+    setCurrentPage(pageNumber);
+    return getCurrentProducts(pageNumber);
+  };
+
+  const handleNextButton = () => {
+    setCurrentPage(currentPage + 1);
+    if (products.length < totalCount) {
+      loadMoreProducts();
+    }
```

Comment:
This we can remove to make the code more simple. We'll let the hook deal with all of this.
Currently, if you go from page 1 to page 2 you load data for page 2. 
If we then go back to page 1 and then to page 2, it will load data for page 3 but paginate to page 2.